### PR TITLE
chore: disable selected sonar rules for integration tests

### PR DESCRIPTION
**Pull request by sissbruecker**
## Description

As an attempt to reduce noise for SonarCloud validation, this change disables some rules, that can cause validation to fail, for integration test modules. Test sources are already treated differently by SonarQube, however for integration tests we also have the test pages / test setups which run through the normal set of checks. As the test setups don't require as high a quality standard as code that gets released, we can drop some of the rules that don't bring any benefit.

Part of #3428

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -129,6 +129,28 @@
                 </pluginManagement>
             </build>
         </profile>
+        <profile>
+            <id>sonar-cloud</id>
+            <properties>
+                <!-- SonarQube rule exclusions for: Integration tests -->
+                <sonar.issue.ignore.multicriteria>e1,e2,e3,e4,e5</sonar.issue.ignore.multicriteria>
+                <!-- Disable: Public types, methods and fields (API) should be documented with Javadoc -->
```
Comment (sissbruecker) [inline pom.xml:137]: Reported as code smell with high severity, however we don't need to document test setups.

File: pom.xml [hunk 2]
```diff
@@ -129,6 +129,28 @@
...
+                <sonar.issue.ignore.multicriteria.e1.ruleKey>java:S1176</sonar.issue.ignore.multicriteria.e1.ruleKey>
+                <sonar.issue.ignore.multicriteria.e1.resourceKey>**/*-integration-tests/**/*.java</sonar.issue.ignore.multicriteria.e1.resourceKey>
+                <!-- Disable: Constructors should only call non-overridable methods -->
```
Comment (sissbruecker) [inline pom.xml:140]: Reported as code smell with high severity. This is a problem in general with Flow, where you often call `Component.add` in the constructor. Unlikely to result in an actual issue in the simple test setup classes that we have.

File: pom.xml [hunk 3]
```diff
@@ -129,6 +129,28 @@
...
+                <sonar.issue.ignore.multicriteria.e2.ruleKey>java:S1699</sonar.issue.ignore.multicriteria.e2.ruleKey>
+                <sonar.issue.ignore.multicriteria.e2.resourceKey>**/*-integration-tests/**/*.java</sonar.issue.ignore.multicriteria.e2.resourceKey>
+                <!-- Disable: String literals should not be duplicated -->
```
Comment (sissbruecker) [inline pom.xml:143]: Reported as code smell with high severity. Certainly a good practice in general, but not as relevant for test code IMO.

File: pom.xml [hunk 4]
```diff
@@ -129,6 +129,28 @@
...
+                <sonar.issue.ignore.multicriteria.e3.ruleKey>java:S1192</sonar.issue.ignore.multicriteria.e3.ruleKey>
+                <sonar.issue.ignore.multicriteria.e3.resourceKey>**/*-integration-tests/**/*.java</sonar.issue.ignore.multicriteria.e3.resourceKey>
+                <!-- Disable: Inheritance tree of classes should not be too deep -->
```
Comment (sissbruecker) [inline pom.xml:146]: Reported as code smell with high severity. We have some test base classes that exceed this limit, but I don't see a lot of potential for issues here.

File: pom.xml [hunk 5]
```diff
@@ -129,6 +129,28 @@
...
+                <sonar.issue.ignore.multicriteria.e4.ruleKey>java:S110</sonar.issue.ignore.multicriteria.e4.ruleKey>
+                <sonar.issue.ignore.multicriteria.e4.resourceKey>**/*-integration-tests/**/*.java</sonar.issue.ignore.multicriteria.e4.resourceKey>
+                <!-- Disable: Standard outputs should not be used directly to log anything -->
```
Comment (sissbruecker) [inline pom.xml:149]: Reported as code smell with high severity. We only have a few occurrences of this, but I think it's OK to just log something in a demo-like page.

**Review by tomivirkki — APPROVED** [review]
The build for this PR doesn't include any Java files https://sonarcloud.io/code?id=vaadin_flow-components&pullRequest=3500 Didn't find a way to test this but the configuration seems to be correct according to the documentation.

**Pull request merged**